packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -120,45 +158,128 @@ contract L2Genesis is Deployer {
+        runWithOptions({
+            _mode: OutputMode.ALL,
+            _fork: LATEST_FORK,
+            _populateNetworkConfig: true,
+            _l1Dependencies: artifactDependencies()
+        });
     }
 
     /// @notice This is used by new experimental interop deploy tooling.
     function runWithEnv() public {
         //  The setUp() is skipped (since we insert a custom DeployConfig, and do not use Artifacts)
         deployer = makeAddr("deployer");
-        runWithOptions(
-            OutputMode.NONE,
-            Config.fork(),
-            L1Dependencies({
+        runWithOptions({
+            _mode: OutputMode.NONE,
+            _fork: Config.fork(),
+            _populateNetworkConfig: false,
+            _l1Dependencies: L1Dependencies({
                 l1CrossDomainMessengerProxy: payable(vm.envAddress("L2GENESIS_L1CrossDomainMessengerProxy")),
                 l1StandardBridgeProxy: payable(vm.envAddress("L2GENESIS_L1StandardBridgeProxy")),
                 l1ERC721BridgeProxy: payable(vm.envAddress("L2GENESIS_L1ERC721BridgeProxy"))
             })
-        );
+        });
     }
 
     /// @notice This is used by foundry tests to enable the latest fork with the
     ///         given L1 dependencies.
     function runWithLatestLocal(L1Dependencies memory _l1Dependencies) public {
-        runWithOptions(OutputMode.NONE, LATEST_FORK, _l1Dependencies);
+        runWithOptions({
+            _mode: OutputMode.NONE,
+            _fork: LATEST_FORK,
+            _populateNetworkConfig: true,
+            _l1Dependencies: _l1Dependencies
+        });
     }
 
     /// @notice Build the L2 genesis.
-    function runWithOptions(OutputMode _mode, Fork _fork, L1Dependencies memory _l1Dependencies) public {
+    /// @param _mode The mode to run the script in.
+    /// @param _fork The fork to build the genesis for.
+    /// @param _populateNetworkConfig If true, the L1 Block contract will be populated with network specific
+    ///                                configuration. Otherwise, the standard genesis will be built.
+    function runWithOptions(
+        OutputMode _mode,
+        Fork _fork,
+        bool _populateNetworkConfig,
+        L1Dependencies memory _l1Dependencies
+    )
+        public
+    {
         console.log("L2Genesis: outputMode: %s, fork: %s", _mode.toString(), _fork.toString());
         vm.startPrank(deployer);
         vm.chainId(cfg.l2ChainID());
 
         dealEthToPrecompiles();
         setPredeployProxies();
-        setPredeployImplementations(_l1Dependencies);
+        setPredeployImplementations();
         setPreinstalls();
         if (cfg.fundDevAccounts()) {
             fundDevAccounts();
         }
         vm.stopPrank();
 
+        // TODO: remove the networkConfig struct and encapsulate this logic in a function to
+        // be shared with the similar logic in Setup.L2()
+        if (_populateNetworkConfig) {
+            networkConfig.l1ChainID = cfg.l1ChainID();
```

Comment:
Can we encapsulate this into an internal function to keep this function smaller?

Reply:
Will do.